src/gill/TrouserPatternTwo.java

```diff
@@ -0,0 +1,378 @@
+
+
+
+    /* Implement abstract methods from super class */
+    @Override
+    protected EMethod assignMethod()
+    {
+        return EMethod.GILL;
+    }
+
+    @Override
+    protected EGarment assignGarment()
+    {
+        return EGarment.TROUSER;
+    }
+
+    @Override
+    protected void addEasement()
+    {
+
+    }
+
+    public TrouserPatternTwo(Measurements dataStore)
+    {
+        if (!readMeasurements(dataStore)) return;
+        addEasement();
+
+        // Create the blocks
+        createBlocks();
+    }
+
+    @Override
+    protected boolean readMeasurements(Measurements dataStore)
+    {
+        try
+        {
+            // Based on measurements for this pattern we can read the following from the scan
+            optSmallofBackWaist = dataStore.getMeasurement("A02").value;
+            hipCircum = dataStore.getMeasurement("A03").value;
+            waistToHipLength = dataStore.getMeasurement("A15").value;
+            kneeCircumR = dataStore.getMeasurement("A18").value;
+            ankleCircumR = dataStore.getMeasurement("A19").value;
+            frontWaistArc = dataStore.getMeasurement("A26").value;
+            backWaistArc = dataStore.getMeasurement("A27").value;
+            frontAbdomenArc = dataStore.getMeasurement("A28").value;
+            backSeatArc = dataStore.getMeasurement("A30").value;
+            frontHipArc = dataStore.getMeasurement("A31").value;
+            backHipArc = dataStore.getMeasurement("A32").value;
+            waistToAbdomen = dataStore.getMeasurement("A33").value;
+            waistToSeat = dataStore.getMeasurement("A34").value;
+            sideStreamUpliftR = dataStore.getMeasurement("A37").value;
+            bodyRise = dataStore.getMeasurement("A38").value;
+            seatDepth = dataStore.getMeasurement("A39").value;
+            hipDepth = dataStore.getMeasurement("A40").value;
+            ankleCircumHeightR = dataStore.getMeasurement("A41").value;
+            kneeCircumHeightR = dataStore.getMeasurement("A42").value;
+            crotchHeight = dataStore.getMeasurement("A43").value;
+            frontSeatArc = dataStore.getMeasurement("A46").value;
+            seatBKZ = dataStore.getMeasurement("A51").value;
+            frontWaistZ = dataStore.getMeasurement("A52").value;
+            backWaistZ = dataStore.getMeasurement("A53").value;
+            waistWidth = dataStore.getMeasurement("A59").value;
+            hipWidth = dataStore.getMeasurement("A60").value;
+            frontAbdomenZ = dataStore.getMeasurement("A61").value;
+            fullCrotchLength = dataStore.getMeasurement("A62").value;
+            // Get name
+            userName = dataStore.getName();
+
+            return true;
+        }
+        catch(MeasurementNotFoundException e)
+        {
+            addMissingMeasurement(dataStore.getName(), e.getMeasurementId());
+            return false;
+        }
+    }
+
+    private void readjustMeasurement()
```

Comment:
What is this? I don't like seeing hard-coded stuff, especially not measurements, that what the measurement store class was for.